qfieldsync/core/cloud_api.py

```diff
@@ -122,13 +130,73 @@ def from_reply(reply: QNetworkReply) -> Optional[CloudException]:
     return CloudException(reply, Exception(message))
 
 
+class CloudAuthMethod(Enum):
+    NONE = 0
+    CREDENTIALS = 1
+    SSO = 2
+
+
+def build_oauth2_auth_config(
+    auth_data: dict[str, Any],
```

Comment:
```suggestion
    auth_data: Dict[str, Any],
```
Unfortunately we support some more ancient Python versions and we should use `Dict` instead of `dict` in QFieldSync :/